### PR TITLE
Authorization flow changes

### PR DESCRIPTION
**Pull request by ruwany**
on behalf of Ayyoob

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java [hunk 1]
```diff
@@ -349,45 +371,37 @@ private void publishDynamicEventStream(String deviceType, String tenantDomain){
...
-            log.error("publishing dynamic event receiver is failed due to  " + e.getMessage(), e);
-        } catch (IOException e) {
-            log.error("publishing dynamic event receiver is failed due to  " + e.getMessage(), e);
-        } catch (JWTClientException e) {
-            log.error("publishing dynamic event receiver is failed due to  " + e.getMessage(), e);
-        } catch (UserStoreException e) {
-            log.error("publishing dynamic event receiver is failed due to  " + e.getMessage(), e);
         } catch (ParseException e) {
-            log.error("publishing dynamic event receiver is failed due to  " + e.getMessage(), e);
+            e.printStackTrace();
```
Comment (sumedharubasinghe) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:392]: Please log this error.

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java [hunk 2]
```diff
@@ -349,45 +371,37 @@ private void publishDynamicEventStream(String deviceType, String tenantDomain){
...
             for (String streamContent:streamList) {
                 JSONParser jsonParser = new JSONParser();
                 JSONObject steamJson = (JSONObject)jsonParser.parse(streamContent);
                 String name = (String) steamJson.get("name");
                 String version = (String) steamJson.get("version");
                 String streamId = name +":"+version;
-                if(eventStreamAdminServiceStub.getStreamDefinitionAsString(streamId)==null) {
-
+                if (eventStreamAdminServiceStub.getStreamDefinitionDto(streamId) == null) {
+                    log.error("its not there");
```
Comment (sumedharubasinghe) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:385]: Make this meaningful log !!
eg:
"Stream Id : " +streamId not found.

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java [hunk 3]
```diff
@@ -349,45 +371,37 @@ private void publishDynamicEventStream(String deviceType, String tenantDomain){
...
                     eventStreamAdminServiceStub.addEventStreamDefinitionAsString(streamContent);
+                } else {
+                    log.error("its there");
```
Comment (sumedharubasinghe) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:388]: Make this meaningful as well. See top comment. Also add context aware details to make this useful in a log file.

**Review by sumedharubasinghe — CHANGES_REQUESTED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 1 above)
Comment (ayyoob) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:392]: this fix was pushed to cloud branch and those changes are not in here, have to get those from the cloud branch : https://github.com/wso2/carbon-device-mgt/pull/715/files

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 3 above)
Comment (ayyoob) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:388]: this fix was pushed to cloud branch and those changes are not in here, have to get those from the cloud branch : https://github.com/wso2/carbon-device-mgt/pull/715/files

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 2 above)
Comment (ayyoob) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:385]: this fix was pushed to cloud branch and those changes are not in here, have to get those from the cloud branch : https://github.com/wso2/carbon-device-mgt/pull/715/files

Comment (sumedharubasinghe) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:385]: What do u mean by not in here? it's in the PR submitted!!

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 3 above)
Comment (sumedharubasinghe) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:388]: What do u mean by not in here? it's in the PR submitted!!

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 1 above)
Comment (sumedharubasinghe) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:392]: What do u mean by not in here? it's in the PR submitted!!

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 3 above)
Comment (ruwany) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:388]: Fix added in PR : https://github.com/wso2/carbon-device-mgt/pull/715

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java (hunk 1 above)
Comment (ruwany) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/admin/DeviceTypePublisherAdminServiceImpl.java:392]: Fix added in PR : https://github.com/wso2/carbon-device-mgt/pull/715

**Review by sumedharubasinghe — APPROVED** [review]

**Pull request merged**